libfwupd/fwupd-self-test.c

```diff
@@ -95,6 +95,61 @@ fwupd_enums_func (void)
 }
 
 static void
+fwupd_remote_http_func (void)
+{
+	gboolean ret;
+	g_autoptr(FwupdRemote) remote = NULL;
+	g_autoptr(GError) error = NULL;
+
+	/* check fallback to HTTP */
+	remote = fwupd_remote_new ();
+	fwupd_remote_set_id (remote, "lvfs");
+	ret = fwupd_remote_load_from_data (remote,
+					   "[fwupd Remote]\n"
+					   "Enabled=true\n"
+					   "Url=https://s3.amazonaws.com/lvfsbucket/downloads/firmware.xml.gz\n",
+					   &error);
+	g_assert_no_error (error);
+	g_assert (ret);
+	g_assert_cmpint (fwupd_remote_get_kind (remote), ==, FWUPD_REMOTE_KIND_HTTP);
+	g_assert_cmpint (fwupd_remote_get_priority (remote), ==, 0);
+	g_assert (fwupd_remote_get_enabled (remote));
+	g_assert (fwupd_remote_get_uri (remote) != NULL);
+	g_assert (fwupd_remote_get_uri_asc (remote) != NULL);
+	g_assert_cmpstr (fwupd_remote_get_filename (remote), ==, "lvfs-firmware.xml.gz");
+	g_assert_cmpstr (fwupd_remote_get_filename_asc (remote), ==, "lvfs-firmware.xml.gz.asc");
+	g_assert (g_str_has_suffix (fwupd_remote_get_filename_cache (remote),
+				    "/var/lib/fwupd/remotes.d/lvfs/metadata.xml.gz"));
+}
+
+static void
+fwupd_remote_local_func (void)
+{
+	gboolean ret;
+	g_autoptr(FwupdRemote) remote = NULL;
+	g_autoptr(GError) error = NULL;
+
+	/* check local repo */
+	remote = fwupd_remote_new ();
+	fwupd_remote_set_id (remote, "local");
+	ret = fwupd_remote_load_from_data (remote,
```

Comment:
Similarly here, you can load the sample "local" data.  Then either add `fwupd_remote_set_enabled` or modify the keyfile in memory in the test suite to mark enabled.